examples/sort-axis.rs

```diff
@@ -145,3 +175,140 @@ fn main() {
+            [75597.83203125, 0.000013488292198201972],
+            [75597.7099609375, 0.000010685034125296037],
+            [75597.587890625, 0.000015045350094244463],
+            [75597.4658203125, 0.0000279214009662375],
+            [75597.34375, 0.000048481399072164026],
+            [75597.2216796875, 0.00007447021310326175],
+            [75597.099609375, 0.00010301909934402759],
+            [75596.9775390625, 0.00012966452075647097],
+            [75596.85546875, 0.00014788235886162946],
+            [75596.7333984375, 0.00015580725716317814],
+            [75596.611328125, 0.00015594598642112382],
+            [75596.4892578125, 0.00014845363317307893],
+            [75596.3671875, 0.00013282540335243705],
+            [75596.2451171875, 0.00011003430347581507],
+            [75596.123046875, 0.00008269157415880566],
+            [75596.0009765625, 0.000053884509965182293],
+            [75595.87890625, 0.000025593452075824607],
+            [75595.7568359375, -1.999206066625981e-6],
+            [75595.634765625, -0.000029234497458252378],
+            [75595.5126953125, -0.00005453591847226998],
+            [75595.390625, -0.0000737870884951274],
+            [75595.2685546875, -0.00008424573434306313],
+            [75595.146484375, -0.00008827030579785403],
+            [75595.0244140625, -0.00009056230744688112],
+            [75594.90234375, -0.00009469997277589672],
+            [75594.7802734375, -0.00010218540321681528],
+            [75594.658203125, -0.00011260114290907028],
+            [75594.5361328125, -0.00012407496245048883],
+            [75594.4140625, -0.00013353166158357196],
+            [75594.2919921875, -0.00013910694259349109],
+            [75594.169921875, -0.00014564817663029577],
+            [75594.0478515625, -0.00015792401974395093],
+            [75593.92578125, -0.00017521854866310149],
+        ];
+        let mut sorted_f = a.column(0).to_vec();
+        sorted_f.sort_by(|a, b| a.partial_cmp(b).unwrap());
+        assert_eq!(sorted_f[0], 75593.92578125);
+        assert_eq!(sorted_f[49], 75599.9072265625);
+
+        let perm = a.sort_axis_by(Axis(0), |i, j| a[[i, 0]] < a[[j, 0]]);
+        let b = a.permute_axis(Axis(0), &perm);
+        assert_eq!(b[[0, 0]], 75593.92578125);
+        assert_eq!(b[[49, 0]], 75599.9072265625);
+    }
+
+    #[test]
+    fn test_long_sort_axis_simple_version() {
+        let a =
+            read_array_data(&test_data_path("array_to_sort.dat"), (283104, 2)).unwrap();
+        // sort first column with vector sort_by
+        let mut sorted_f = a.column(0).to_vec();
+        sorted_f.sort_by(|a, b| a.partial_cmp(b).unwrap());
+        assert_eq!(sorted_f[0], 74880.05859375);
+        assert_eq!(sorted_f[283103], 109439.94140625);
+
+        let perm = a.sort_axis_by(Axis(0), |i, j| a[[i, 0]] < a[[j, 0]]);
+        assert_eq!(perm.indices[0], 5897);
+        assert_eq!(perm.indices[283103], 176939);
+
+        let b = permute_axis_simple(a, Axis(0), &perm);
```

Comment:
if you want to compare permute_axis_simple and permute_axis - why isn't everything else in the function equal?

Reply:
Good point, that was left over from some earlier experimentation.  Just pushed a commit so that the test setups are identical.